app/uk/gov/hmrc/agentsubscriptionfrontend/GuiceModule.scala

```diff
@@ -35,6 +36,45 @@ class GuiceModule(environment: Environment, configuration: Configuration) extend
   override protected lazy val mode: Mode = environment.mode
   override protected lazy val runModeConfiguration: Configuration = configuration
 
+  trait ConfigPropertyType[A] {
+    def bindConfigProperty(clazz: Class[A])(propertyName: String): ScopedBindingBuilder
+  }
+
+  object ConfigPropertyType {
+    implicit val stringConfigProperty = new ConfigPropertyType[String] {
+      def bindConfigProperty(clazz: Class[String])(propertyName: String): ScopedBindingBuilder =
+        bind(clazz).annotatedWith(Names.named(s"$propertyName")).toProvider(new StringConfigPropertyProvider(propertyName))
+
+      private class StringConfigPropertyProvider(propertyName: String) extends Provider[String] {
```

Comment:
This class is very similar to the `IntConfigPropertyProvider` below on line 61.
It seems at first glance the only difference between the two being is the `Configuration.getString/getInt` call and whether the class deals with ints or strings. 
Could a generic class be used instead? e.g. parameterized with either int/string types and provided with a Configuration function to call. 
I'm guessing it might end up looking too complicated to gain any real benefit though..